pennylane/measurements/sample.py

```diff
@@ -106,4 +109,49 @@ def circuit(x):
             )
         wires = Wires(wires)
 
-    return MeasurementProcess(Sample, obs=op, wires=wires)
+    return _Sample(Sample, obs=op, wires=wires)
+
+
+# TODO: Make public when removing the ObservableReturnTypes enum
+class _Sample(SampleMeasurement):
+    """Measurement process that returns the samples of a given observable."""
+
+    def process_samples(
+        self, samples: Sequence[complex], shot_range: Tuple[int] = None, bin_size: int = None
+    ):
+        name = self.obs.name if self.obs is not None else None
+        # Select the samples from samples that correspond to ``shot_range`` if provided
+        if shot_range is not None:
+            # Indexing corresponds to: (potential broadcasting, shots, wires). Note that the last
+            # colon (:) is required because shots is the second-to-last axis and the
+            # Ellipsis (...) otherwise would take up broadcasting and shots axes.
+            samples = samples[..., slice(*shot_range), :]
+
+        if len(self.wires) != 0:
+            # if wires are provided, then we only return samples from those wires
+            samples = samples[..., self.wires]
+
+        num_wires = samples.shape[-1]  # wires is the last dimension
+
+        if self.obs is None:
+            # if no observable was provided then return the raw samples
+            return samples if bin_size is None else samples.reshape(num_wires, bin_size, -1)
+
+        if name in {"PauliX", "PauliY", "PauliZ", "Hadamard"}:
+            # Process samples for observables with eigenvalues {1, -1}
+            samples = 1 - 2 * qml.math.squeeze(samples)
```

Comment:
Here, `samples` starts as an array with shape `(shots, num_wires)`. Won't this produce another array with the same shape?

Reply:
Same confusion as the other comments 👍